centos_ci.sh

```diff
@@ -22,79 +22,155 @@ set -e
+  mkdir /tmp/glide
+  tar --directory=/tmp/glide -xvf glide-${GLIDE_TAG}-${GLIDE_OS_ARCH}.tar.gz
+  export PATH=$PATH:/tmp/glide/${GLIDE_OS_ARCH}
+}
 
-set +x
-# For PR build, GIT_BRANCH is set to branch name other than origin/master
-if [[ "$GIT_BRANCH" = "origin/master" ]]; then
+function prepare() {
+  install_and_setup_golang;
+  setup_repo;
+  setup_glide;
+
+  echo "CICO: Preparation complete"
+}
+
+function run_tests() {
+  cd $GOPATH/src/github.com/minishift/minishift
+  make clean test cross fmtcheck prerelease
+  # Run integration test with 'kvm' driver
+  MINISHIFT_VM_DRIVER=kvm make integration
+
+  echo "CICO: Tests ran successfully"
+}
+
+function artifacts_upload_on_pr_and_master_trigger() {
+  set +x
+  # For PR build, GIT_BRANCH is set to branch name other than origin/master
+  if [[ "$GIT_BRANCH" = "origin/master" ]]; then
+    # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
+    mkdir -p minishift/master/$BUILD_NUMBER/
+    cp -r out/* minishift/master/$BUILD_NUMBER/
+    RSYNC_PASSWORD=$1 rsync -a --relative minishift/master/$BUILD_NUMBER/ minishift@artifacts.ci.centos.org::minishift/
+    echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/master/$BUILD_NUMBER ."
+  else
+    # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
+    mkdir -p minishift/pr/$ghprbPullId/
+    cp -r out/* minishift/pr/$ghprbPullId/
+    RSYNC_PASSWORD=$1 rsync -a --relative minishift/pr/$ghprbPullId/ minishift@artifacts.ci.centos.org::minishift/
+    echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/pr/$ghprbPullId ."
+  fi
+}
+
+function docs_artifacts_upload() {
+  set +x
+
+  version=$(cat docs/build/variables.adoc | cut -d' ' -f2)
+  mkdir -p minishift/docs/$(version)
+  cp docs/build/minishift-adocs.tar minishift/docs/$(version)/
+  ln -s $(pwd)/minishift/docs/$(version) minishift/docs/latest
   # http://stackoverflow.com/a/22908437/1120530; Using --relative as --rsync-path not working
-  mkdir -p minishift/master/$BUILD_NUMBER/
-  cp -r out/* minishift/master/$BUILD_NUMBER/
-  RSYNC_PASSWORD=$PASS rsync -a --relative minishift/master/$BUILD_NUMBER/ minishift@artifacts.ci.centos.org::minishift/
-  echo "Find Artifacts here http://artifacts.ci.centos.org/minishift/minishift/master/$BUILD_NUMBER ."
+  RSYNC_PASSWORD=$1 rsync -a --relative minishift/docs minishift@artifacts.ci.centos.org::minishift/
+  echo "Find docs artifacts here http://artifacts.ci.centos.org/minishift/minishift/docs/$(version)."
+}
+
+# Execution begins here
+load_jenkins_vars;
+install_core_deps;
+
+# Required for artifacts upload
+PASS=$(echo $CICO_API_KEY | cut -d'-' -f1-2)
+
+echo "JOB_NAME: $JOB_NAME"
+
+# Intentionally using wrong condition to test docs artifacts uploads
+if [[ "$JOB_NAME" != "minishift-docs" ]]; then
+  install_docker;
+  prepare;
+  cd $GOPATH/src/github.com/minishift/minishift
+  make gen_adoc_tar
```

Comment:
@budhrg can you try:

    make gen_adoc_tar UID=1000

This should overwrite the 0. Question is whether it will work when the files are created.